Makefile.am

```diff
@@ -220,15 +193,13 @@ clean-local::
 maintainer-clean-local::
 	rm -rf dist/ node_modules/ package-lock.json
 
-node_modules/.stamp: package.json
-	@echo ''
-	@echo '    package.json changed'
-	@echo ''
-	@echo '    Please run:'
-	@echo ''
-	@echo '        tools/npm-install'
-	@echo ''
-	@false
+FORCE:
```

Comment:
Is that some leftover from experimentation? If that's supposed to land, it needs a comment at least. There is no other reference than the dependency of package-lock.json, and it's neither phony nor ever built.

Reply:
This is a farly standard idiom: https://www.gnu.org/software/make/manual/html_node/Force-Targets.html

I'll add a comment, though.

Reply:
I still don't see it. package-lock.json is not phony conceptually -- it is a real file that make should build. What's wrong with just `package-lock.json:` ?

Reply:
`package-lock.json:` without any dependencies is a rule for producing `package-lock.json` if it doesn't exist (it can never be out of date, because it has no dependencies).

`package-lock.json: FORCE` is a rule that will run every single time someone asks for package-lock.json to be updated.  Our script is run every time in order to check that `node_modules` is correctly in sync with the gitlink.

Reply:
Ah, I see, thanks! (So this actually *is* like a phony target)

Reply:
Not exactly.  The timestamp of the file is still considered in this case: in the event that we replace the file with a new version, the targets depending on it will get rebuilt.

Pure phony targets never stat the underlying "file".